src/k8s/pkg/snap/snap.go

```diff
@@ -329,7 +330,25 @@ func (s *snap) SnapctlSet(ctx context.Context, args ...string) error {
 }
 
 func (s *snap) PreInitChecks(ctx context.Context, config types.ClusterConfig) error {
-	// TODO: check for available ports for k8s-dqlite, apiserver, containerd, etc
+	// TODO(eac): Make these configurable or part of a Config struct
+	ports := []int{
+		config.APIServer.GetSecurePort(),    // kubeAPIServerPort
+		config.Datastore.GetK8sDqlitePort(), // k8sDqlitePort
+		10248,                               // kubeletHealthzPort
```

Comment:
All these ports can be set in the service extra args. We need to check from there for the right values.